### ISSUE TITLE
20190709 franz.hex makes system unusable

### ISSUE THREAD

**Issue by mikrosk** [atarijookie/ce-atari#206]
OK, finally I had time and motivation to investigate #199 further. It's not an update process bug, it's really bad firmware. When I use the latest stretch image and update everything I get into the state described in #199:

- atari keyboard and mouse is randomly freezing
- usb keyboard switched into the console (LCTRL+LALT+RALT+RCTRL) is randomly switching back
- franz version shows 0000 in CE_CONF
- CE_CONF reports connection lost every 3-4 seconds

As soon as I reflash franz.hex from 20170413, all issues are gone. There must be some serious bug hidden in the new version. Also from memory this looks pretty consistent with other user reports about CE being unusable after an update.

My config: 
![IMG_20191015_192439](https://user-images.githubusercontent.com/3062260/66854576-e39c4600-ef81-11e9-9674-ae9b05c08aac.jpg)

**Comment by mikrosk** [comment]
Btw that 'update not needed' is of course a ticking time bomb because I have locally overwritten franz.version & franz.hex so the next time an update is available, git will loudly complain that it can't update the working directory because of local changes.

**Comment by mikrosk** [comment]
Looking at the log for franz firmware, I see there's a lot of floppy-related commits but also a few dangerously looking ones like "refactoring" or "making XY faster/better" ;-)

**Comment by mikrosk** [comment]
An update on this one, this must be somehow Falcon related. During Silly Venture 2019 we were updating some CE 3.0 units and I could see the exact same behaviour. So it's not my faulty (v2) unit, it's some Franz FW bug.

**Comment by drummerjoe** [comment]
> An update on this one, this must be somehow Falcon related. During Silly Venture 2019 we were updating some CE 3.0 units and I could see the exact same behaviour. So it's not my faulty (v2) unit, it's some Franz FW bug.

Any chance you can provide a downgrade process for v1 users?  I'd be willing to test this entire process but my device is totally hosed at the moment from attempting to upgrade.

![IMG_20200124_163949](https://user-images.githubusercontent.com/49952774/73106044-78d7a080-3ec8-11ea-80b9-52b763473d82.jpg)

**Comment by mikrosk** [comment]
As improbable as it sounds, this has been an user error the whole time. Namely, my error. ;) The original 34-pin floppy cable wiring in Falcon is actually the opposite of the usual HD floppy drives' standard so when I installed Gotek in my Falcon, I made a cable which is the opposite to the Falcon/ST standard and used that the whole time. CE naturally respect the Falcon/ST standard so one doesn't have to twist the ribbon cable but in my case that meant that I have been connecting the cable upside down the whole time!

After twisting the cable I can use the latest version of everything:

![IMG_20200202_203116](https://user-images.githubusercontent.com/3062260/73614001-7e856280-45fb-11ea-909f-0db8a38a2253.jpg)

- we found this by pure coincidence, Expi has had a Falcon with similar "upgrade", i.e. PC/Gotek compatible and we compared it to stock Falcon at some point
- update process from stretch 2017 to stretch 2019 is very, very non-intuitive - `CE_CONF` assumes updating of each component but in the background we're installing `git` and whatnot, it takes forever and when using `CE_CONF` there is no way of knowing what's going on. Only thanks to ssh + `ce_update.sh` I could realise that I have to wait a bit longer
- @drummerjoe I'd suggest downloading the latest image from http://joo.kie.sk/cosmosex/download/ce_raspbian_stretch_2019-08-13.zip and `dd` it to the SD card manually
